### PR TITLE
device - infortrend - add temperature sensor

### PR DESCRIPTION
**Pull request by ervin09**
Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: mibs/IFT-SNMP-MIB
Comment (PipoCanaja) [inline mibs/IFT-SNMP-MIB:1]: Hi @ervin09 
Could you please put this MIB file into the appropriate directory and not in the mibs/ general directory.
Thx

**Comment by murrant** [comment]
@ervin09 if you can at least provide the snmprec by running `./scripts/collect-snmp-data.php -h <hostname> -v <model>` (which can be generated without any extra requirements)

**Comment by PipoCanaja** [comment]
Hi @ervin09 
I fixed the issues in the YAML file (indentation and OID for the hardware and serial).
- Could you check if behaves correctly on your real hadrware ?
- Could you update the snmpsim file with the fixed YAML (so we get hardware populated, seems that it is not currently)
- I'll then re-update the JSON file

And we should be good to merge :)

**Comment by PipoCanaja** [comment]
The "state" part was removed because it cannot be done in YAML. PHP code is necessary to parse the bitmap structure of the states. But Voltage and Temperature should be good to go.

**Comment by ervin09** [comment]
hi @PipoCanaja , thanks for looking into it. yes Temperature and Voltage behave as they should. i've a graph for each and any of those (Voltage 5V, Voltage 12V and Temperature). The other states, i agree it was an issue to be done differently.

**Comment by PipoCanaja** [comment]
> hi @PipoCanaja , thanks for looking into it. yes Temperature and Voltage behave as they should. i've a graph for each and any of those (Voltage 5V, Voltage 12V and Temperature). The other states, i agree it has to be done differently, it shows very little and i do want to have a better overview. i tried the QNAP approach on it, but i didn't get the same result (as you can see) if you can point in a direction that i can adapt any previous PHP codes that are already in Librenms , i can surely open up a new PR with the PHP code added. cheers mate Ervin

The idea is that from a single number, you get multiple flags (either 0 or 1). You could use the existing function `q_bridge_bits2indices($hex_data)` and call it with the HEX value received from the state OID. 
```
    /* convert hex string to an array of 1-based indices of the nonzero bits
     * ie. '9a00' -> '100110100000' -> array(1, 4, 5, 7) 
    */  ```

And then, you can set the all the state sensors from the result of the call (in the example above, bit 1, 4, 5 and 7 are set, so the matching 4 states would be "ON" and all others would be "OFF". 

I did not find a PHP sensors implemented that way yet.

And toe make things more difficult, the INFORTREND MIB also have multibits sub-states , like bits 0 to 2. 
```
ldStatus OBJECT-TYPE
    SYNTAX      Integer32
    MAX-ACCESS  read-only
    STATUS      current
    DESCRIPTION "Logical drive status
                    BITS 0-2 : Status Code (RO):
                              0 : Good
                              1 : Rebuilding
                              2 : Initializing
                              3 : Degraded
                              4 : Dead
                              5 : Invalid
                              6 : Incomplete
                              7 : Drive Missing
                    BITS 3-5 : Reserved.
                    BIT 6 : Logical Drive Data Not Synced
                    BIT 7 : Logical Drive Off-line (RW)."
    ::= { ldEntry 6 }   
```

Basically, you would need to implement this magic into those files : 
`~/includes/discovery/sensors/state/infortrend.inc.php`
`~/includes/polling/sensors/state/infortrend.inc.php`

**Comment by PipoCanaja** [comment]
Would like to do it here or to get this PR merged and open another one later ?

**Review by PipoCanaja — APPROVED** [review]
LGTM if it is OK for you

**Comment by ervin09** [comment]
@PipoCanaja thanks for the advice. And for all the rest , appreciate it a lot. Lets merge this one, ill put some time to try something with the hex values. I will def "bother" you guys again soon, sorry for that :)) and i really appreciate your help 

PS: if you could point me into one more direction for the test cases? (The version ive its an ubuntu vm i downloaded from the main site, with a ready installation of librenms, but building the cases always fails, i should look into this too)

**Comment by murrant** [comment]
I think we've used user_func to parse binary states before...
